pkg/controllers/workspace_controller.go

```diff
@@ -141,6 +156,48 @@ NodeListLoop:
 	return validNodesCount, nil
 }
 
+func (c *WorkspaceReconciler) isVHDInstalled(ctx context.Context, nodeObj *corev1.Node) bool {
+	// check if label accelerator=nvidia exists in the node
+	var foundLabel, foundCapacity bool
+	if nvidiaLabelVal, found := nodeObj.Labels[NvidiaLabelKey]; found {
+		if nvidiaLabelVal == NvidiaLabelValue {
+			klog.InfoS("nvidia accelerator label has been found", "node", nodeObj.Name)
+			foundLabel = true
+		}
+	}
+
+	// check Status.Capacity.nvidia.com/gpu has value
+	capacity := nodeObj.Status.Capacity
+	if capacity != nil && !capacity.Name(CapacityNvidiaGPU, "").IsZero() {
+		klog.InfoS("nvidia GPU capacity value found greater than 0", "node", nodeObj.Name, CapacityNvidiaGPU, capacity.Name(CapacityNvidiaGPU, "").Value())
+		foundCapacity = true
+	}
+
+	if foundLabel && foundCapacity {
+		return true
+	}
+	return false
+}
+
+func (c *WorkspaceReconciler) checkAndInstallDADI(ctx context.Context, nodeObj *corev1.Node) (bool, error) {
+	var installed bool
+	if customLabel, found := nodeObj.Labels[GPUProvisionerCustomLabel]; found {
```

Comment:
Is checking label enough? Should you check the existence of the DS pod?

Reply:
Added. return the DS and check if is available.